todo_app/app.py

```diff
@@ -1,14 +1,57 @@
-from flask import Flask
-
+from flask import Flask, request, render_template
 from todo_app.flask_config import Config
+from todo_app.data.session_items import get_items, add_item, get_item, save_item, delete_item
+from operator import itemgetter
 
 app = Flask(__name__)
 app.config.from_object(Config)
 
 
 @app.route('/')
 def index():
-    return 'Hello World!'
+    items = get_items()
+    ##renders the template with the items sorted by status
+    return render_template('index.html',items=sorted(items,key=itemgetter('status'), reverse=True))
+
+@app.route('/add_todo_item', methods=['POST']) 
+def add_todo_item(): 
+    new_todo_item = request.form.get('newitem', "")
+    add_item(new_todo_item)
+    items = get_items()
+    ##renders the template with the items sorted by status
+    return render_template('index.html',items=sorted(items,key=itemgetter('status'),reverse=True))
+
+@app.route('/completed', methods=['POST']) 
+def completed(): 
+    ##retrieve the id of the completed task
+    completed_item = request.values.get('id')
+    ##retrieve the item of the completed task
+    item = get_item(completed_item)
```

Comment:
`completed_item` should be called `completed_item_id` because it's an id not an item - `item` is the actual item